metabase/metabase
# Inline Date Filters

## Description

This adds date-specific shortcuts to the bulk filter modal. Common shortcut actions are added inline, but all date filters are still available via popover. A few notes

- clicking a selected shortcut should clear the filter
- adding a non-shortcut filter should display it inline as a tag
- clicking a custom date filter's tag should bring up the date picker to modify it
- clearing a custom date filter should restore the shortcut options to visibility

view | image
--- | ---
Shortcut Filter View | <img width="661" alt="Screen Shot 2022-06-21 at 8 31 49 AM" src="https://user-images.githubusercontent.com/30528226/174825588-4ea71756-cbf7-47e9-a9a2-c6a1c81851d6.png">
Custom Filter View | ![Screen Shot 2022-06-21 at 1 20 39 PM](https://user-images.githubusercontent.com/30528226/174880965-47e40285-6dea-4dce-aa4e-11a4e93075ac.png)


## Testing Steps
- Open a table that has a date field (like orders)
- add, modify and remove a variety of date filters
- see that filters added in notebook mode or via the sidebar appear as expected in the modal as well
